Merge the RadMon data extract source code into rrfs_util.

xyzemc opened the pull request:
https://github.com/NOAA-GSL/rrfs_utl/issues/43

The RadMon (radiance monitoring) package is used to extract data from
radiance diagnostic files to generate the binary files for plotting the statistics from radiance data assimilation, and generate tons of images for each satellite instrument and channel.  It has been used in GDAS and NAM DA operational.
This pull request is for adding the RadMon source code into rrfs_util.

The data extract step from RadMon include four packs of source code:
verf_radang.fd/  
verf_radbcoef.fd/  
verf_radbcor.fd/ 
verf_radtime.fd/

All these codes have put in the directory 'radmon' under rrfs_util. The corresponding makefile has been modified and tested.  Four executables will be placed into the ./bin directory as all other code did.

xyzemc committed:
This for merge the radmon data extract source code into rrfs_util.

christopherwharrop-noaa reviewed (changes requested):
There is a lot of duplication in this pull request.  The specification of compiler flags for different compilers should ideally be done in one place for the entire repository, not duplicated verbatim in multiple places.  Additionally, the introduction of multiple `kinds.f90` creates unnecessary duplication, particularly since rrfs_utls already has a kinds module.  Please remove duplication and make use of existing code and build specifications where possible.

hu5970 commented:
@christopherwharrop-noaa This repository is to host many small applications for RRFS. Some of those applications are isolated from others. I agree it is good to reduce the duplication. We also need to check this case by case. "kind.f90" used here may not have the same definition as the kinds module used by other. Do we have a way to keep some code used locally (like kinds.f90 only used by this application)? The same possible is also applying for compiling options.

xyzemc commented:
@christopherwharrop-noaa @hu5970 I will compare the kind.f90 in this PR and the existing kind.f90 in rrfs_util to see if they are same and can avoid the duplication.

xyzemc committed:
Move three duplicate codes (kind.f90, read_diag.f90, sparserr.f90) to a seperate directory.
Modified the CMake file to generate three module (.mod) in the include directory.

hu5970 committed:
Merge branch 'develop' into radmon_test

hu5970 commented:
@xyzemc Please remove save directory and sync with current develop branch.

xyzemc committed:
Delete radmon/save directory

christopherwharrop-noaa commented on radmon/shared/CMakeLists.txt:
```diff
@@ -0,0 +1,25 @@
+add_library(${PROJECT_NAME}::radmonlib ALIAS radmonlib)
+
+target_compile_definitions(radmonlib PRIVATE "_REAL8_")
+#target_link_libraries(radmonlib PUBLIC ${PROJECT_NAME}::ncio)
```
Please remove commented code

christopherwharrop-noaa commented on radmon/verf_radbcoef.fd/CMakeLists.txt:
```diff
@@ -0,0 +1,17 @@
+
+add_executable(radmon_bcoef.x ${SRCS})
+
+#include_directories( /scratch1/BMC/wrfruc/mhu/rrfs/v0.7.1/ufs-srweather-app/include )
```
Please remove commented code

christopherwharrop-noaa commented on radmon/verf_radtime.fd/bad_chan.f90:
```diff
@@ -0,0 +1,94 @@
+      character(60)                     :: fname
+
+
+!      write(*,*) '--> open_bad_chan_file, date, cycle = ', date, cycle
```
This looks like commented code leftover from debugging.  Perhaps it should be removed?

christopherwharrop-noaa commented on radmon/verf_radtime.fd/bad_penalty.f90:
```diff
@@ -0,0 +1,100 @@
+      logical                           :: lexist = .FALSE.
+      character(60)                     :: fname
+      
+!      write(*,*) '--> open_bad_penalty_file, date, cycle = ', date, cycle
```
This looks like commented code leftover from debugging.  Perhaps it should be removed?

christopherwharrop-noaa reviewed (changes requested):
Thank you for taking the time to address my comments.  This looks great!  There are a couple commented lines in `CMakeLists.txt` files that should be removed, and you might also consider removing what look like (I'm not sure) commented out debugging `write` statements.  Once the Cmake comments are removed, I'll approve.

xyzemc committed:
delete comment line.

xyzemc committed:
Deleted the comment lines.
Modified the kinds.mod to kinds_radmon.mod in share/kinds_radmon.F90.
Modified the 'use kinds' to 'use kinds_radmon'

christopherwharrop-noaa reviewed (approved):
Thank you for making those last few updates.
